ingestion/src/metadata/ingestion/source/pipeline/dagster.py

```diff
@@ -0,0 +1,174 @@
+    Source as WorkflowSource,
+)
+from metadata.generated.schema.type.entityLineage import EntitiesEdge
+from metadata.generated.schema.type.entityReference import EntityReference
+from metadata.ingestion.api.source import InvalidSourceException
+from metadata.ingestion.models.pipeline_status import OMetaPipelineStatus
+from metadata.ingestion.source.pipeline.pipeline_service import PipelineServiceSource
+from metadata.utils.connections import (
+    create_and_bind_session,
+    get_connection,
+    test_connection,
+)
+from metadata.utils.helpers import datetime_to_ts
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+STATUS_MAP = {
+    "success": StatusType.Successful.value,
+    "failed": StatusType.Failed.value,
+    "queued": StatusType.Pending.value,
+}
+
+
+class DagsterSource(PipelineServiceSource):
+    """
+    Implements the necessary methods ot extract
+    Pipeline metadata from Dagster's metadata db
+    """
+
+    config: WorkflowSource
+
+    def __init__(
+        self,
+        config: WorkflowSource,
+        metadata_config: OpenMetadataConnection,
+    ):
+        self._session = None
+        self.service_connection = config.serviceConnection.__root__.config
+        self.engine: Engine = get_connection(self.service_connection.connection)
+        super().__init__(config, metadata_config)
+        # Create the connection to the database
+
+    @classmethod
+    def create(cls, config_dict, metadata_config: OpenMetadataConnection):
+        config: WorkflowSource = WorkflowSource.parse_obj(config_dict)
+        connection: DagsterConnection = config.serviceConnection.__root__.config
+        if not isinstance(connection, DagsterConnection):
+            raise InvalidSourceException(
+                f"Expected DagsterConnection, but got {connection}"
+            )
+        return cls(config, metadata_config)
+
+    @property
+    def session(self) -> Session:
+        """
+        Return the SQLAlchemy session from the engine
+        """
+        if not self._session:
+            self._session = create_and_bind_session(self.engine)
+
+        return self._session
+
+    def get_run_list(self):
+
+        run_list = self.session.execute("select * from runs")
+        return run_list
+
+    def yield_pipeline(self, pipeline_details) -> Iterable[CreatePipelineRequest]:
+        """
+        Convert a DAG into a Pipeline Entity
+        :param serialized_dag: SerializedDAG from dagster metadata DB
+        :return: Create Pipeline request with tasks
+        """
+        task_list = [{"name": row["pipeline_name"]} for row in self.get_run_list()]
+        dag = pipeline_details
+        yield CreatePipelineRequest(
+            name=pipeline_details["pipeline_name"],
+            description=dag[3],
+            pipelineUrl=pipeline_details[3],  # Just the suffix
+            pipelineLocation=pipeline_details[3],
+            startDate=dag[9],
```

Comment:
I believe that the `Row` objects returned by SQLAlchemy can return each column by name, not only by index. This would be helpful to identify better what we are trying to retrieve. Something like `pipeline_details.<my_column>` instead of `pipeline_details[magic_number]`.